hack/packages/package-tools/cmd/prepare.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"bytes"
+	_ "embed" // required to embed file
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"text/template"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+
+	"github.com/vmware-tanzu/tanzu-framework/hack/packages/package-tools/utils"
+)
+
+// prepareCmd is for preparing package tooling for building packages
+var prepareCmd = &cobra.Command{
+	Use:   "prepare",
+	Short: "Prepare for building packages",
+	RunE:  runPrepare,
+}
+
+//go:embed config/carvelTools-config.yaml
+var carvelTools []byte
+
+var clean bool
+
+func init() {
+	rootCmd.AddCommand(prepareCmd)
+	prepareCmd.Flags().BoolVar(&clean, "clean", false, "Deletes the existing Carvel tool binaries")
+}
+
+func runPrepare(cmd *cobra.Command, args []string) error {
+	if err := downloadCarvelBinaries(); err != nil {
+		return fmt.Errorf("couldn't download carvel binaries: %w", err)
+	}
+	return nil
+}
+
+// downloadCarvelBinaries downloads the carvel binaries and places them in hack/tools/bin in project root dir
+func downloadCarvelBinaries() error {
+	c := new(CarvelTools)
+	err := yaml.Unmarshal(carvelTools, c)
+	if err != nil {
+		return fmt.Errorf("unmarshal error: %w", err)
+	}
+
+	projectRootDir, err := utils.GetProjectRootDir()
+	if err != nil {
+		return err
+	}
+
+	toolsBinDir := filepath.Join(projectRootDir, "hack", "tools", "bin")
+
+	// clean hack/tools/bin directory
+	if clean {
+		if _, err := os.Stat(toolsBinDir); !os.IsNotExist(err) {
+			if err := os.RemoveAll(toolsBinDir); err != nil {
+				return fmt.Errorf("couldn't delete carvel binaries: %w", err)
+			}
+		}
+	}
+
+	if err := utils.CreateDir(toolsBinDir); err != nil {
+		return err
+	}
+
+	for _, tool := range c.Tools {
```

Comment:
optional: we could use concurrency to speed this up, but also okay if you want do it later.

Reply:
sure, we can use concurrency here, but right now I think it's not worth it as we are downloading just 4 binaries